### PR TITLE
@jkhales/add apiv2 option credentials

### PR DESCRIPTION
**Pull request by jkhales**
This was already approved here: https://github.com/expo/expo-cli/pull/1431
but github wouldn't let me reopen it.

Was able to remove some of the ios code and make some of the code "android only" due to @quinlanj recent work.

ran:
```
expo build:ios
expo build:android
expo build:ios --clear-credentials
expo build:android --clear-credentials
```

### CODE REVIEW

File: packages/xdl/src/credentials/IosCredentials.ts [hunk 1]
```diff
@@ -100,19 +102,38 @@ async function getExistingUserCredentials(
   appleTeamId: string,
   type: string
 ): Promise<CredsList> {
-  const { err, certs } = await Api.callMethodAsync('getExistingUserCredentials', [], 'post', {
-    username,
-    appleTeamId,
-    type,
-  });
+  let certs;
```
Comment (quinlanj) [inline packages/xdl/src/credentials/IosCredentials.ts]: looks like `getExistingUserCredentials` is only used by `getExistingPushKeys` and `getExistingDistCerts` which are not used anymore so u might be able to get rid of all of them

File: packages/xdl/src/credentials/Credentials.ts [hunk 2]
```diff
@@ -47,65 +48,114 @@ export async function credentialsExistForPlatformAsync(
...
-    platform,
-    ...metadata,
-  });
-
-  if (err) {
-    throw new Error('Error deleting credentials.');
+  if (process.env.EXPO_NEXT_API) {
+    const user = await UserManager.ensureLoggedInAsync();
+    const api = ApiV2.clientForUser(user);
+    console.log('deleting android credentials');
```
Comment (quinlanj) [inline packages/xdl/src/credentials/Credentials.ts]:
```suggestion

```

File: packages/xdl/src/credentials/IosCredentials.ts (hunk 1 above)
Comment (jkhales) [reply packages/xdl/src/credentials/IosCredentials.ts]: Nice catch.

**Comment by jkhales** [comment]
> would feel more assured if u have time to test for the commands that use it
> (ie)
> -`expo fetch` `expo upload` for `getCredentialsForPlatform`
> 
> * `expo google-play`(?) for `updateCredentialsForPlatform`
> * `expo build:android` for `updateCredentialsForPlatform` and `removeCredentialsForPlatform`

```
fetch:ios:certs
fetch:android:keystore
fetch:android:hashes
fetch:android:upload-cert

fetch:android:upload-cert
opt-in-google-play-signing
```

Still have to test 
```
upload:ios
upload:android
```
Waiting on play store.
But since this is only activate with a flag am going to merge this to get as much internal testing in the meantime.

**Pull request merged**